### PR TITLE
Fix issues with github-workflows

### PR DESCRIPTION
**Pull request by odasko**
### Issues
(1) Jobs/steps requiring access to the registry fail when the user of the reusable workflow has their registry set to _internal_

> Example: aquasecurity/trivy-action in run-security-scans.yml returns the following error: Screenshot 2022-11-17 at 12 39 42

(2) check-github-security job in run-security-scans.yml fails with exitcode 1 regardless of whether or not there are high or critical code scanning alerts in Github Security code scanning tab. 
Screenshot 2022-11-17 at 12 37 26

(3) run-terraform.yml is still using som versions of Github Actions with the depricated 'set-output' command

### Solutions
(1) Log into registry with the docker/login-action before using commands/actions which require access to the registry

> Example: Grant aquasecurity/trivy-action access to internal packages by logging in to registry and pulling image before the github action is run (as suggested in https://support.github.com/ticket/personal/0/1871892).

(2) Fix issue with check-github-security job in run-security-scans. check-github-security now checks the Github security code scanning alerts corresponding to the current github.ref (i.e. the branch or pr that is running the workflow)
(3) Update hashicorp/setup-terraform versions in run-terraform.yml

### Tested here https://github.com/kartverket/shipwreck/actions/runs/3489543049 and here https://github.com/kartverket/shipwreck/actions/runs/3497264671/jobs/5856167141

### CODE REVIEW

File: .github/workflows/run-security-scans.yml [hunk 1]
```diff
@@ -79,22 +84,39 @@ jobs:
...
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker pull image
+        run: docker pull $IMAGE_URL
+
       - name: Run Trivy vulnerability scanner
-        uses: aquasecurity/trivy-action@master
+        uses: aquasecurity/trivy-action@9ab158e8597f3b310480b9a69402b419bc03dbd5
```
Comment (anderssonw) [inline .github/workflows/run-security-scans.yml:119]: Kan vi bruke release her og?

File: .github/workflows/run-security-scans.yml [hunk 2]
```diff
@@ -172,15 +194,19 @@ jobs:
...
+          severity_check=$(curl -s -H "Accept: application/vnd.github+json" -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" https://api.github.com/repos/${{ github.repository }}/code-scanning/alerts?ref=${{github.ref}} | jq '[.[] | select(.state == "open" and (.rule.security_severity_level == "high" or .rule.security_severity_level == "critical"))] | any')
           if [[ $severity_check == 'true' ]]
           then
-            echo "Error: High or critical vulnerabilities detected. Go to the Code Scanning section of the Github Security-tab to review these vulnerabilities."
-            exit 1
+            echo "Error: High or critical vulnerabilities detected. Go to the Code Scanning section of the Github Security-tab to review these vulnerabilities. If this is a pull request, search for 'is:open pr:<pr-number>'";
+            exit 1;
```
Comment (anderssonw) [inline .github/workflows/run-security-scans.yml:201]: Fikk du denne til å funke til slutt? hva var greia?

Comment (anderssonw) [reply .github/workflows/run-security-scans.yml:201]: Not sure I follow but if it works sick 💪

File: .github/workflows/run-security-scans.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ on:
       # The image reference generated by a build job.
       image_url:
         description: "An image_url of the form registry/repository:tag."
```
Comment (anderssonw) [inline .github/workflows/run-security-scans.yml:8]: Husk å oppdatere README, og kanskje legge til at dersom man skal bruke Trivy scanning må man legge ved Image (selv om det kanskje er innlysende).

**Review by anderssonw — APPROVED** [review]

File: .github/workflows/run-security-scans.yml (hunk 3 above)
Comment (anderssonw) [reply .github/workflows/run-security-scans.yml:8]: Godkjenner uansett, flisespikkeri :)

**Review by KronenE — APPROVED** [review]

**Pull request merged**